api/filtering/db_filters.py

```diff
@@ -321,7 +325,10 @@ def query_filters(
         sp_filter, host_type_filter = _system_profile_filter(filter)
         filters += sp_filter
     if staleness:
-        filters += _staleness_filter(staleness, host_type_filter)
+        if identity:
+            filters += _staleness_filter(staleness, host_type_filter, identity)
+        else:
+            filters += _staleness_filter(staleness, host_type_filter, identity)
```

Comment:
Both paths do the same thing
```suggestion
        filters += _staleness_filter(staleness, host_type_filter, identity)
```